packages/core/src/awsService/appBuilder/serverlessLand/metadataManager.ts

```diff
@@ -0,0 +1,122 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as nodefs from 'fs' // eslint-disable-line no-restricted-imports
+import { ToolkitError } from '../../../shared/errors'
+
+interface IaC {
+    id: string
+    name: string
+}
+interface Runtime {
+    id: string
+    name: string
+    version: string
+}
+interface PatternData {
+    name: string
+    description: string
+    runtimes: Runtime[]
+    iac: IaC[]
+}
+
+export interface ProjectMetadata {
+    patterns: Record<string, PatternData>
+}
+
+/**
+ * Manages metadata for serverless application patterns
+ * Implements Singleton pattern to ensure single instance across the application
```

Comment:
You can probably drop this second comment, it's just describing what a singleton is